account_bank_receipt/models/account_bank_receipt.py

```diff
@@ -0,0 +1,378 @@
+    line_ids = fields.One2many(
+        comodel_name='account.move.line',
+        related='move_id.line_ids',
+        string='Lines',
+        readonly=True,
+    )
+    company_id = fields.Many2one(
+        comodel_name='res.company',
+        required=True,
+        states={'done': [('readonly', '=', True)]},
+        default=lambda self: self.env['res.company']._company_default_get(),
+    )
+    total_amount = fields.Float(
+        compute='_compute_bank_receipt',
+        readonly=True,
+        store=True,
+        digits=dp.get_precision('Account'),
+    )
+    check_count = fields.Integer(
+        string='Number of Checks',
+        compute='_compute_bank_receipt',
+        readonly=True,
+        store=True,
+    )
+    is_reconcile = fields.Boolean(
+        string='Reconcile',
+        compute='_compute_bank_receipt',
+        readonly=True,
+        store=True,
+    )
+    partner_id = fields.Many2one(
+        comodel_name='res.partner',
+        domain="[('customer', '=', True)]",
+        required=True,
+        states={'done': [('readonly', True)]},
+    )
+    receipt_line = fields.One2many(
+        comodel_name='account.bank.receipt.line',
+        inverse_name='order_id',
+        states={'done': [('readonly', True)]},
+    )
+
+    @api.constrains('receipt_line')
+    def _check_allocation(self):
+        for rec in self:
+            payments = rec.receipt_line.mapped('move_line_ids')
+            for payment in payments:
+                allocation = sum(rec.receipt_line.filtered(
+                    lambda l: l.move_line_ids.id == payment.id
+                    ).mapped('allocation'))
+                residual = payment.amount_residual
+                if allocation > residual:
+                    raise ValidationError(_(
+                        '%s has sum Allocation > Residual'
+                        ) % (payment.move_id.name))
+
+    def unlink(self):
+        for rec in self:
+            if rec.state == 'done':
+                raise UserError(
+                    _("The deposit '%s' is in valid state, so you must "
+                      "cancel it before deleting it.") % rec.name)
+        return super(AccountBankReceipt, self).unlink()
+
+    @api.multi
+    def backtodraft(self):
+        for rec in self:
+            if rec.move_id:
+                rec.move_id.button_cancel()
+                rec.move_id.line_ids.filtered(
+                    lambda x: x.account_id.reconcile).remove_move_reconcile()
+                rec.move_id.unlink()
+            rec.write({'state': 'draft'})
+        return True
+
+    @api.model
+    def create(self, vals):
+        if vals.get('name', '/') == '/':
+            vals['name'] = self.env['ir.sequence'].\
+                with_context(ir_sequence_date=vals.get('receipt_date')).\
+                next_by_code('account.bank.receipt')
+        return super(AccountBankReceipt, self).create(vals)
+
+    @api.model
+    def _prepare_account_move_vals(self, deposit):
```

Comment:
Should use 'bank_transit' as var name instead of deposit ? (you can consult with p'Lek)